src/binxml/name.rs

```diff
@@ -22,16 +22,16 @@ impl<'a> BinXmlName<'a> {
         BinXmlName(Cow::Borrowed(s))
     }
 
-    pub fn from_binxml_stream<'c>(
+    pub fn from_binxml_stream(
         cursor: &mut Cursor<&'a [u8]>,
-        ctx: Context<'a, 'c>,
+        chunk: Option<&'a EvtxChunk<'a>>,
     ) -> Result<BinXmlName<'a>, Error> {
         // Important!!
         // The "offset_from_start" refers to the offset where the name struct begins.
         let name_offset = try_read!(cursor, u32);
 
         // If name is cached, read it and seek ahead if needed.
-        if let Some((name, _, n_bytes_read)) = ctx.cached_string_at_offset(name_offset) {
+        if let Some((name, _, n_bytes_read)) = chunk.and_then(|chunk| chunk.string_cache.get_string_and_hash(name_offset)) {
```

Comment:
Nice use of `and_then`, but I think the “why” of utilizing the option here isn’t clear enough.
If we were to merge the chunk and the parsing context, than the context part could not be optional (we always need to have the parsing settings available)